src/Common/AzureAvsCostCalculator.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+using Azure.Migrate.Export.Models;
+
+namespace Azure.Migrate.Export.Common
+{
+    public class AzureAvsCostCalculator
+    {
+        //AVS datasets
+        private Dictionary<AssessmentInformation, AVSAssessmentPropertiesDataset> AvsAssessmentsData;
+
+        private bool IsCalculated;
+
+        private double TotalAvsComputeCost;
+
+        public AzureAvsCostCalculator()
+        {
+            AvsAssessmentsData = new Dictionary<AssessmentInformation, AVSAssessmentPropertiesDataset>();
+
+            IsCalculated = false;
+
+            TotalAvsComputeCost = 0.00;
+        }
+
+        public bool IsCalculationComplete()
+        {
+            return IsCalculated;
+        }
```

Comment:
IsCalculated is never set to true, is this working?

Reply:
public void Calculate()
        {
            CalculateSqlIaaSCost();
            CalculateWebAppIaaSCost();
            CalculateVMIaaSCost();

            TotalIaaSComputeCost = SqlIaaSComputeCost + WebAppIaaSComputeCost + VmIaaSComputeCost;
            TotalIaaSStorageCost = SqlIaaSStorageCost + WebAppIaaSStorageCost + VmIaaSStorageCost;
            TotalIaaSSecurityCost = SqlIaaSSecurityCost + WebAppIaaSSecurityCost + VmIaaSSecurityCost;
            TotalIaaSAhubSavings = SqlIaaSAhubSavings + WebAppIaaSAhubSavings + VmIaaSAhubSavings;

            managementTargetCount = vmserverProdRowCount + sqlProdRowCount + webappProdRowCount;

            IsCalculated = true;
        }

Reply:
yes, this is working fine, because the function gets called once for an entire run. Will set the bool to true though.